### ISSUE TITLE
Missing documentation on adding elevation data to routing data

### ISSUE THREAD

**Issue by Svantulden** [graphhopper/graphhopper#309]
Hi,

First of all, I want to say that I love this routing engine. For my use case it does a lot of things that other engines can't (routing via custom shapefile network that is converted to OSM, having many profiles active at the same time, easily extendable. All those things weren't possible in OSRM, which I looked at first). 

Now I want to add elevation data to my routing data, though I am having trouble how to do that exactly. I saw in the `config.properties` that I can specify a provider and cache directory, but what else is needed for elevation data support? Also, I saw that a CGIAR provider is supported also ( #204 ), which should be better error-corrected elevation data? 

So basically my questions are: 

- Which provider should I use for elevation data? Is CGIAR the best way to go because of it's correctness? 
- What settings do I need to change to support it in Graphhopper? What do I download and where to store it? 

I'm willing to contribute to the documentation when I know how to do it. Thanks!

**Comment by karussell** [comment]
Thanks for the kind words! And indeed this needs to have better docs. The CGIAR is preferred but is in general not public domain, although we requested a license for our and our users usage: https://graphhopper.com/public/license/CGIAR.txt

The only necessary configuration change is: 
`graph.elevation.provider=cgiar`
or
`graph.elevation.provider=srtm`

then the default cache directory (/tmp/srtm) will be used. For large areas it is highly recommended to use a SSD disc, thus you need to specify the cache directory:
`graph.elevation.cachedir=/myssd/ele_cache/`

> What do I download and where to store it? 

All should work automatically. Another setting is to specify the location where the files are downloaded - e.g. if the servers are not reachable, then you set:
`graph.elevation.baseurl`
E.g. for CGIAR there are two URLs I found:
`http://droppr.org/srtm/v4.1/6_5x5_TIFs`
`http://srtm.csi.cgiar.org/SRT-ZIP/SRTM_V41/SRTM_Data_GeoTiff/`
where the last one is only accessibly if you specify the [full zip file](srtm.csi.cgiar.org/SRT-ZIP/SRTM_V41/SRTM_Data_GeoTiff/srtm_01_02.zip)

If the area is small and you need a faster import you can change the default MMAP setting to:
`graph.elevation.dataaccess=RAM_STORE`

**Comment by Svantulden** [comment]
Awesome, thanks for your comment and adding the docs! Sorry I've been busy lately, will try it soon!

**Comment by benchuk** [comment]
Hi, joining Svantulden, love the library.
I am having trouble loading the DTM (elevation) to android.
I added the flag to config.properties
graph.flagEncoders=mtb
graph.elevation.provider=srtm
than pushed the created data folder to android (where I am loading all the data....)
pushed the following files.
edges,geometry,location_index,names,nodes,nodes_ch_fastest_mtb,properties,shortcuts_fastest_mtb
routing works fine, but when debugging... response.PointList.elevations is empty...
should I push more data to android? missing any steps?
Thanks, ben.

**Comment by karussell** [comment]
When changing the configs then a re-import is necessary.

(Please move this discussion to the mailing list)

**Comment by benchuk** [comment]
How do I join the mailing list?

**Comment by karussell** [comment]
See the readme or https://graphhopper.com/#community

**Comment by benchuk** [comment]
Thanks.
